docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -581,6 +581,18 @@ Canisters pay for their cycles which makes them inherently vulnerable to attacks
 
 Consider monitoring, early authentication, rate limiting on canister level to mitigate this. Also, be aware that an attacker will aim for the call consuming most cycles. See the "Cycle balance drain attacks section" in [how to audit an Internet Computer canister](https://www.joachim-breitner.de/blog/788-How_to_audit_an_Internet_Computer_canister).
 
+### Do not rely on ingress message inspection
```

Comment:
Nice, thanks for adding this @tmu0 ! LGTM. 

A minor point: Would it make sense to link this section from the section "Make sure any action that only a specific user should be able to do requires authentication" and vice versa?